hatchery/nextflow.go

```diff
@@ -343,9 +343,31 @@ func createNextflowUserResources(userName string, bucketName string, batchComput
 				"Resource": [
 					"%s"
 				]
-			}
+			},
+			{
+				"Effect": "Allow",
+            			"Action": [
+                			"batch:RegisterJobDefinition"
+            			],
+            			"Resource": [
+					"arn:aws:batch:*:*:job-definition/*",
+            			],
+            			"Condition": {
+                			"StringEquals": {
+                    				"batch:User": [
+                        				"%s"
+                    				],
```

Comment:
interesting, what does this part achieve (user condition)?

Reply:
`batch:User` was assigned to `nobody` in [this]([url](https://docs.aws.amazon.com/batch/latest/userguide/ExamplePolicies_BATCH.html#iam-example-job-def)) template -- I'm just testing to see if this can be user-specific using username!

Reply:
right but this policy is already assigned to a single user, so i'm wondering why we need this

Reply:
hmm you may be right, we may not need this afterall